ampd/src/broadcaster.rs

```diff
@@ -165,14 +167,16 @@ where
             .change_context(Error::TxBuilding)?;
 
         self.estimate_gas(sim_tx).await.map(|gas| {
-            let gas_adj = (gas as f64 * self.config.gas_adjustment) as u64;
+            let gas_adj = gas as f64 * self.config.gas_adjustment;
 
             Fee::from_amount_and_gas(
                 Coin {
-                    amount: (gas_adj as f64 * self.config.gas_price.amount).ceil() as u128,
+                    amount: cast((gas_adj.mul(self.config.gas_price.amount)).ceil())
```

Comment:
yeah neither `into()` nor `try_into()` work. The `mul` was indeed to get around clippy lint, clippy doesn't really flags multiplication of two floats, because any "overflowing" value is set to `f64::INFINITY` which is a valid value. In this case it's flagged because `amount` is a custom struct even though it behaves like a float `pub struct FiniteAmount(f64)`, so I just basically delegated the lint checking to the `mul` implementation